golem-worker-executor-base/src/worker.rs

```diff
@@ -355,6 +352,31 @@ impl<Ctx: WorkerCtx> Worker<Ctx> {
         self.stop_internal(false).await;
     }
 
+    pub async fn stop_if_idle(&self) -> bool {
+        let running_guard = self.running.lock().await;
+        match running_guard.as_ref() {
+            Some(running) => {
+                if is_running_worker_idle(running) {
+                    if self.stopping.compare_exchange(
+                        false,
+                        true,
+                        Ordering::Acquire,
+                        Ordering::Relaxed,
+                    ) == Ok(false)
+                    {
+                        self.stop_internal_running(running_guard, false).await;
```

Comment:
probably works, just interested: what is the for no-race-condition guarantee between
- is_running_worker_idle
- stopping mutex
- stop_internal_running?

and is there no way that the stop can fail? as in stopping a non-idle one

Reply:
I'm going to document this better in the next PR